integration-tests/tests/src/tests/sync/flexible.ts

```diff
@@ -1397,6 +1399,19 @@ describe.skipIf(environment.missingServer, "Flexible sync", function () {
             expect(subs).to.have.length(1);
             expect([...subs][0].queryString).to.equal("age > 10");
           });
+
+          it("returns true and removes a subscription with an empty name", async function (this: RealmContext) {
+            addSubscription(this.realm, this.realm.objects(FlexiblePersonSchema.name).filtered("age > 10"));
+            const { subs } = addSubscriptionForPerson(this.realm, { name: "" });
+            expect(subs).to.have.length(2);
+
+            await subs.update((mutableSubs) => {
+              expect(mutableSubs.removeByName("")).to.be.true;
+            });
+
+            expect(subs).to.have.length(1);
+            expect([...subs][0].queryString).to.equal("age > 10");
```

Comment:
This would be equivalent and perhaps a bit simpler?

```suggestion
            const [ sub ] = subs;
            expect(sub.queryString).to.equal("age > 10");
```

Reply:
That can be a small but good refactor we could add in a separate PR (perhaps for our test refactor ticket), because the previous pattern is also used in other places so would be good to change all of them together.